packages/commonwealth/client/scripts/views/Sublayout.tsx

```diff
@@ -32,27 +38,54 @@ const Sublayout = ({
     };
   }, [forceRerender]);
 
+  useEffect(() => {
+    let timer;
+    if (resizing) {
+      timer = setTimeout(() => {
+        setResizing(false);
+      }, 200); // adjust delay as needed
+    }
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [resizing]);
+
   useEffect(() => {
     if (
       localStorage.getItem('dark-mode-state') === 'on' &&
       localStorage.getItem('user-dark-mode-state') === 'on'
     ) {
       document.getElementsByTagName('html')[0].classList.add('invert');
     }
+
+    const onResize = () => {
+      if (userToggledVisibility === null) {
+        setMenu({ name: 'default', isVisible: !isWindowSmallInclusive });
+      }
+    };
+
+    window.addEventListener('resize', onResize);
+
+    return () => {
+      window.removeEventListener('resize', onResize);
+    };
   }, []);
 
   const chain = app.chain ? app.chain.meta : null;
   const terms = app.chain ? chain.terms : null;
   const banner = app.chain ? chain.communityBanner : null;
-  const showSidebar = menuVisible || !isWindowSmallInclusive;
 
   return (
     <div className="Sublayout">
       <div className="header-and-body-container">
         <SublayoutHeader onMobile={isWindowSmallInclusive} />
         <div className="sidebar-and-body-container">
-          {showSidebar && <Sidebar isInsideCommunity={hasCommunitySidebar} />}
-          <div className="body-and-sticky-headers-container">
+          <Sidebar isInsideCommunity={hasCommunitySidebar} />
+          <div
+            className={`body-and-sticky-headers-container 
+            ${menuVisible ? 'menu-visible' : 'menu-hidden'} 
+            ${resizing ? 'resizing' : ''}`}
```

Comment:
can we use `clsx` for conditional classes? it is way more readable. We use it already in the project

The code would look like this
```
          <div
            className={clsx('body-and-sticky-headers-container', {
              'menu-visible': menuVisible,
              'menu-hidden': !menuVisible,
              resizing,
            })}
          >
```